src/main/java/com/mycompany/shop/security/SecurityController.java

```diff
@@ -0,0 +1,75 @@
+package com.mycompany.shop.security;
+
+import com.mycompany.shop.basket.BasketRepository;
+import com.mycompany.shop.security.userdetails.UserDetailsImpl;
+import com.mycompany.shop.user.User;
+import com.mycompany.shop.user.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/public/auth")
+public class SecurityController {
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    BasketRepository basketRepository;
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @PostMapping("/login")
+    public ResponseEntity<?> authenticateUser(@RequestBody User user) {
+        Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(user.getUsername(), user.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+
+        User authenticatedUser = new User(userDetails.getId(), userDetails.getUsername());
+
+        return ResponseEntity.ok(authenticatedUser);
```

Comment:
here you need to get basket for this user and keep it in the session bean

Reply:
+